### PR TITLE
Add create change set endpoint

### PR DESCRIPTION
**Pull request by alexmorken**

### CODE REVIEW

File: moto/cloudformation/responses.py [hunk 1]
```diff
@@ -77,6 +77,50 @@ def create_stack(self):
...
+        stack_name = self._get_param('StackName')
+        change_set_name = self._get_param('ChangeSetName')
+        stack_body = self._get_param('TemplateBody')
+        template_url = self._get_param('TemplateURL')
+        role_arn = self._get_param('RoleARN')
+        update_or_create = self._get_param('ChangeSetType', 'CREATE')
+        parameters_list = self._get_list_prefix("Parameters.member")
+        tags = dict((item['key'], item['value'])
+                    for item in self._get_list_prefix("Tags.member"))
+        # Hack dict-comprehension
```
Comment (terrycain) [inline moto/cloudformation/responses.py]: There a reason why we cant use regular dictionary comprehensions?

File: moto/cloudformation/responses.py [hunk 2]
```diff
@@ -250,6 +294,17 @@ def list_exports(self):
 </UpdateStackResponse>
 """
 
+CREATE_CHANGE_SET_RESPONSE_TEMPLATE = """<CreateStackResponse>
+  <CreateChangeSetResult>
+    <Id>{{change_set_id}}</Id>
+    <StackId>{{ stack_id }}</StackId>
+  </CreateChangeSetResult>
+  <ResponseMetadata>
+    <RequestId>b9b4b068-3a41-11e5-94eb-example</RequestId>
```
Comment (terrycain) [inline moto/cloudformation/responses.py]: Look [here](https://github.com/spulec/moto/blob/81f3cbb54880bdf68fd6f52b9aa30ea25a4b7bd9/moto/cloudwatch/responses.py#L18) and [here](https://github.com/spulec/moto/blob/81f3cbb54880bdf68fd6f52b9aa30ea25a4b7bd9/moto/cloudwatch/responses.py#L195) for returning a proper request ID. If your up to it, see if the other methods can be done too

File: moto/cloudformation/utils.py [hunk 3]
```diff
@@ -10,6 +10,11 @@ def generate_stack_id(stack_name):
     return "arn:aws:cloudformation:us-east-1:123456789:stack/{0}/{1}".format(stack_name, random_id)
 
 
+def generate_changeset_id(changeset_name):
+    random_id = uuid.uuid4()
+    return 'arn:aws:cloudformation:us-east-1:123456789:changeSet/{0}/{1}'.format(changeset_name, random_id)
```
Comment (terrycain) [inline moto/cloudformation/utils.py]: Cloudwatch should have region specific ARNs right?

**Comment by terrycain** [comment]
Great contribution, welcome to Moto. PR looks good, but I think theres a few places where we could be a bit more flexible.

File: moto/cloudformation/responses.py (hunk 1 above)
Comment (alexmorken) [reply moto/cloudformation/responses.py]: I was just following convention in the file.  I would be happy to change it to a real dict comprehension.

File: moto/cloudformation/utils.py (hunk 3 above)
Comment (alexmorken) [reply moto/cloudformation/utils.py]: Yes - I was again just copying convention used in `generate_stack_id` above.  Trying to be predictable.

File: moto/cloudformation/responses.py (hunk 1 above)
Comment (terrycain) [reply moto/cloudformation/responses.py]: Yeah swap it out, we dont need to support py26 anymore and py27 supports it if I remember correctly.

File: moto/cloudformation/utils.py (hunk 3 above)
Comment (terrycain) [reply moto/cloudformation/utils.py]: Fair enough, if your up for fixing all of them to use regions, by all means ;-)

File: moto/cloudformation/responses.py (hunk 2 above)
Comment (alexmorken) [reply moto/cloudformation/responses.py]: Cool - thanks - will take a look at it!

**Comment by alexmorken** [comment]
I will start working on execute_change_set and should hopefully have a PR up for that in a few.

File: moto/cloudformation/responses.py (hunk 1 above)
Comment (alexmorken) [reply moto/cloudformation/responses.py]: Yea - that's what I was thinking - probably some 2.6 cruft.

**Comment by alexmorken** [comment]
Everything should addressed per your feedback.

**Comment by alexmorken** [comment]
I made another small change. I set the status of the stack to `REVIEW_IN_PROGRESS` if `create_change_set` created the stack initially.

**Comment by alexmorken** [comment]
Ok - everything is looking good at this point and also a bit more true to how AWS acts, too.

**Comment by terrycain** [comment]
:D Looks great

**Pull request merged**